akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterShardingSettings.scala

```diff
@@ -131,6 +129,72 @@ object ClusterShardingSettings {
   private[akka] def roleOption(role: String): Option[String] =
     if (role == "") None else Option(role)
 
+  final class PassivationStrategySettings(
+      val strategy: String,
+      val idleTimeout: FiniteDuration,
+      val leastRecentlyUsedLimit: Int) {
+
+    def withIdleStrategy(timeout: FiniteDuration): PassivationStrategySettings =
+      copy(strategy = "idle", idleTimeout = timeout)
+
+    def withLeastRecentlyUsedStrategy(limit: Int): PassivationStrategySettings =
+      copy(strategy = "least-recently-used", leastRecentlyUsedLimit = limit)
+
+    private def copy(
+        strategy: String,
+        idleTimeout: FiniteDuration = idleTimeout,
+        leastRecentlyUsedLimit: Int = leastRecentlyUsedLimit): PassivationStrategySettings =
+      new PassivationStrategySettings(strategy, idleTimeout, leastRecentlyUsedLimit)
+  }
+
+  object PassivationStrategySettings {
+    def apply(config: Config): PassivationStrategySettings = {
+      val settings = new PassivationStrategySettings(
+        strategy = config.getString("passivation.strategy").toLowerCase,
```

Comment:
I don't think it's needed for know strings but in other places we use toRootLowerCase

Reply:
Ok, switched to `toRootLowerCase` anyway